invenio_records_resources/resources/files/resource.py

```diff
@@ -62,6 +66,7 @@ def create_url_rules(self):
             route("GET", routes["list"], self.search),
             route("GET", routes["item"], self.read),
             route("GET", routes["item-content"], self.read_content),
+            route("GET", routes["item-archive"], self.read_archive),
```

Comment:
```suggestion
        if self.config.allow_archive_download:
            url_rules += [
                route("GET", routes["item-archive"], self.read_archive),
            ]
```

This is the same as the `self.config.allow_upload` below. That way the resource/view functioanlity stays a bit more simple and the enabling/disabling happens on a higher-level.

Reply:
I obviously didn't understand what to do. This is the FileResourceConfig that doesn't contain the setting. And here I can't access `current_app`. Please add some more explanation!